src/MassTransit.Abstractions/Internals/Extensions/TaskExtensions.cs

```diff
@@ -105,17 +105,32 @@ async Task<T> WaitAsync()
 
             async Task WaitAsync()
             {
-                var delayTask = Task.Delay(Debugger.IsAttached ? Timeout.InfiniteTimeSpan : timeout, cancellationToken);
-
-                var completed = await Task.WhenAny(task, delayTask).ConfigureAwait(false);
-                if (completed == delayTask)
+                var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                try
                 {
-                    task.IgnoreUnobservedExceptions();
+                    Task delayTask = Task.Delay(Debugger.IsAttached ? Timeout.InfiniteTimeSpan : timeout, cts.Token)
```

Comment:
`Debugger.IsAttached` P/Invokes on every use, is this really needed here?
For DEV-debugging 
```c#
#if DEBUG
Task delayTask = Task.Delay(Timeout.InfiniteTimeSpan, cts.Token);
#else
Task delayTask = Task.Delay(timeout, cts.Token);
#endif
```
is more performant (at least in production).

Reply:
I agree